Baseline Crossgen2 outer loop failing tests

nattress opened the pull request:
The outer loop Crossgen2 pipeline is now being monitored for failures by our vendor team and needs to run clean. Some of the remaining failing tests are run in special modes we don't enable during crossgen2 (ie, ObjectStackAllocationTests would need COMPlus_JitObjectStackAllocation setting during crossgen, superpmicollect uses an alt-jit).

The remaining failures have issues listed in the project file for further investigation since they look like Crossgen2 bugs.

nattress commented:
/azp list

nattress commented:
/azp run runtime-coreclr crossgen2 outerloop

nattress commented:
/azp run runtime-coreclr crossgen2

trylek reviewed (approved):
Frankly speaking I think it would be cleaner to have them in issues.targets but that requires changes to build scripts. Feel free to merge your change in at least as a stop-gap solution and I'll clean this up at some point by adding the test build mode property indicator to XUnit wrapper build.

nattress committed:
Exclude coreclr tests in crossgen2 mode

Add a `TestBuildMode` property that can be used to exclude tests that fail in pipelines that use Crossgen / Crossgen2 compilation.

Baseline known test failures from the outerloop crossgen2 pipeline.

nattress commented:
I added support to issues.targets so we can exclude tests as @trylek suggested.
